projects/igniteui-angular/src/lib/services/overlay/overlay.ts

```diff
@@ -559,30 +561,38 @@ export class IgxOverlayService implements OnDestroy {
-      viewContainerRefOrSettings?: ViewContainerRef | OverlaySettings,
-      moduleRefOrSettings?: OverlaySettings): OverlaySettings {
-        let result: OverlaySettings | undefined = moduleRefOrSettings;
-        if (viewContainerRefOrSettings && !(viewContainerRefOrSettings instanceof ViewContainerRef)) {
-            result = viewContainerRefOrSettings;
-        }
-        return result;
-    }
-
+    /**
+     * Creates overlayInfo. Sets the info's `elementRef`, `componentRef`and `settings`. Also
+     * initialize info's `ngZone`, `transformX` and `transformY`.
+     * @param component ElementRef or Type. If type is provided dynamic component will be created
+     * @param viewContainerRefOrSettingsOrCreate (optional): If ElementRef is provided for `component` this
+     * parameter is OverlaySettings. Otherwise it could be ViewContainerRef or OverlayCreateSettings and will be
+     * used when dynamic component is created.
+     * @param settings (optional): OverlaySettings when `component` is not ElementRef.
+     * @returns OverlayInfo
+     */
     private getOverlayInfo(
         component: ElementRef | Type<any>,
-        viewContainerRefOrSettings?: ViewContainerRef | OverlaySettings): OverlayInfo | null {
+        viewContainerRefOrSettingsOrCreate?: ViewContainerRef | OverlaySettings | OverlayCreateSettings,
+        settings?: OverlaySettings): OverlayInfo | null {
         const info: OverlayInfo = { ngZone: this._zone, transformX: 0, transformY: 0 };
         if (component instanceof ElementRef) {
             info.elementRef = component;
+            const overlaySettings = viewContainerRefOrSettingsOrCreate as OverlaySettings;
+            info.settings = Object.assign({}, this._defaultSettings, overlaySettings); 
         } else {
             let dynamicComponent: ComponentRef<any>;
-            if (viewContainerRefOrSettings instanceof ViewContainerRef) {
-                dynamicComponent = viewContainerRefOrSettings.createComponent(component);
+            if (viewContainerRefOrSettingsOrCreate instanceof ViewContainerRef) {
+                const viewContainerRef = viewContainerRefOrSettingsOrCreate as ViewContainerRef;
+                dynamicComponent = viewContainerRef.createComponent(component);
+                info.settings = Object.assign({}, this._defaultSettings, settings); 
             } else {
                 const environmentInjector = this._appRef.injector;
-                dynamicComponent = createComponent(component, { environmentInjector });
+                const createSettings = viewContainerRefOrSettingsOrCreate as OverlayCreateSettings | undefined;
+                const elementInjector = createSettings?.injector;
+                dynamicComponent = createComponent(component, { environmentInjector, elementInjector });
                 this._appRef.attachView(dynamicComponent.hostView);
+                info.settings = Object.assign({}, this._defaultSettings, createSettings); 
```

Comment:
this line repeat 3 times - no need. Just `let` setting as `OverlaySettings`, default to the 3rd param if you want, resolve in cases and assign once?

Reply:
Also in the cases w/ OverlayCreateSettings, should we:
```ts
const { elementInjector, ...overlaySettings } = createSettings;
```
To remove the injector from the stored info object? We should be disposing info-s correctly so not likely a risk, but still. Unf, not sure if the settings not matching by ref anymore could be something to worry about.